src/core/server/core_app/core_app.ts

```diff
@@ -52,7 +52,13 @@ export class CoreApp {
     const httpSetup = coreSetup.http;
     const router = httpSetup.createRouter('/');
     router.get({ path: '/', validate: false }, async (context, req, res) => {
-      const defaultRoute = await context.core.uiSettings.client.get<string>('defaultRoute');
+      let defaultRoute = await context.core.uiSettings.client.get<string>('defaultRoute');
+      // TODO: [RENAMEME] Temporary code for backwards compatibility.
+      // https://github.com/opensearch-project/OpenSearch-Dashboards/issues/334
+      defaultRoute =
+        defaultRoute === '/app/kibana_overview'
+          ? '/app/opensearch_dashboards_overview'
+          : defaultRoute;
```

Comment:
Can we add any tests for this logic?

Reply:
Sure, just will update it still, clicking on the "make it my default page" makes it this specific route but users can manually update that route to include like '/app/kibana_overview#' so will update it and add some quick tests.